kafkaserver/server.go

```diff
@@ -271,12 +282,18 @@ func (c *connection) readMessage() error {
 }
 
 func (c *connection) handleMessage(message []byte) error {
-	if !c.authContext.Authorised && c.s.cfg.KafkaServerListenerConfig.AuthenticationType == auth.AuthenticationTLS {
+	if !c.authContext.Authenticated && c.s.cfg.KafkaServerListenerConfig.AuthenticationType == auth.AuthenticationTLS {
 		if err := c.authoriseWithClientCert(); err != nil {
 			return err
 		}
 	}
-	return protocol.HandleRequestBuffer(message, &NewHandler{conn: c}, c.conn)
+	apiKey := int16(binary.BigEndian.Uint16(message))
+	authType := c.s.cfg.KafkaServerListenerConfig.AuthenticationType
+	authenticated := authType == "" || apiKey == protocol.APIKeyAPIVersions || apiKey == protocol.APIKeySaslHandshake || apiKey == protocol.APIKeySaslAuthenticate || c.authContext.Authenticated
+	if !authenticated {
+		return errors.Errorf("cannot handle Kafka apiKey: %d as connection as authentication type is %s but connection has not been authenticated", apiKey, authType)
```

Comment:
```suggestion
		return errors.Errorf("cannot handle Kafka apiKey: %d as authentication type is %s but connection has not been authenticated", apiKey, authType)
```

Reply:
@purplefox I think you missed this small one.

Reply:
fixed now